js/bootstrap-datepicker.js

```diff
@@ -1196,6 +1200,10 @@
 			}
 
 			if (!target.hasClass('disabled')){
+				// Allow clicking on elements inside a day
+				if (target.parent().hasClass('day'))
+			        target = target.parent();
```

Comment:
Since you allow html, you code won't work if I send `<div><div></div></div>`, you should change `e.target` by `e.currentTarget`, but then you should do a full test to be sure it does not break everything.